### PR TITLE
add authentication for mqtt client

### PR DESCRIPTION
**Pull request by MP91**
- use baseimage and cli in sdk

## Checklist - Manual tasks

* [ ] Examples are executing successfully
* [ ] Created/updated unit tests. Code Coverage percentage on new code shall be >= 80%.
* [ ] Created/updated integration tests.
* [ ] Devcontainer can be opened successfully
* [ ] Devcontainer can be opened successfully behind a corporate proxy
* [ ] Devcontainer can be re-built successfully
* [ ] Extended the documentation (e.g. README.md, CONTRIBUTING.md, Velocitas)

### CODE REVIEW

File: sdk/src/sdk/pubsub/MqttPubSubClient.cpp [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
 #include "sdk/middleware/Middleware.h"
 
+#include "mqtt/connect_options.h"
 #include <mqtt/async_client.h>
```
Comment (BjoernAtBosch) [inline sdk/src/sdk/pubsub/MqttPubSubClient.cpp:25]: Project-external includes shall use angle brackets:
```suggestion
#include <mqtt/async_client.h>
#include <mqtt/connect_options.h>
```

File: sdk/src/sdk/pubsub/MqttPubSubClient.cpp [hunk 2]
```diff
@@ -35,9 +36,41 @@ class MqttPubSubClient : public IPubSubClient, private mqtt::callback {
...
+                     const std::string& trustStorePath, const std::string& keyStorePath,
+                     const std::string& privateKeyPath)
         : m_client{brokerUri, clientId} {
         m_client.set_callback(*this);
+        auto sslopts = mqtt::ssl_options_builder()
+                           .trust_store(trustStorePath)
+                           .key_store(keyStorePath)
+                           .private_key(privateKeyPath)
+                           .error_handler([](const std::string& msg) {
+                               std::cerr << "SSL Error: " << msg << std::endl;
```
Comment (BjoernAtBosch) [inline sdk/src/sdk/pubsub/MqttPubSubClient.cpp:68]:
```suggestion
                               logger().error("SSL Error: {}", msg);
```

File: sdk/include/sdk/IPubSubClient.h [hunk 3]
```diff
@@ -50,6 +50,52 @@ class IPubSubClient {
...
                                                          const std::string& clientId);
 
+    /**
+     * @brief Create a new instance of an MQTT client connecting to a broker at the specified
+     * address using the provided credentials
+     *
+     * @param brokerUri address of the MQTT broker to connect to
+     * @param clientId used to identify the client at the MQTT broker
+     * @param username username to get access to the MQTT broker
+     * @param password password to get access to the MQTT broker
```
Comment (BjoernAtBosch) [inline sdk/include/sdk/IPubSubClient.h:60]:
```suggestion
     * @param username to get access to the MQTT broker
     * @param password to get access to the MQTT broker
```

File: .velocitas.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{
+    "packages": [
+        {
+            "name": "devenv-runtimes",
+            "version": "v2.2.6"
```
Comment (BjoernAtBosch) [inline .velocitas.json:5]: Why do we need the runtimes here?

**Review by BjoernAtBosch — CHANGES_REQUESTED** [review]

File: .velocitas.json (hunk 4 above)
Comment (MP91) [reply .velocitas.json:5]: We already had it before (removed the install dapr script, mqtt broker config in settings,...), so I replaced the old style with the cli

Comment (BjoernAtBosch) [reply .velocitas.json:5]: Yes, your're right. Wasn't aware of that any more ...

**Review by BjoernAtBosch — DISMISSED** [review]
LGTM

**Pull request merged**